File: programmers/algorithm/level2/sung-silver/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int front = 0;
        int end = people.length-1;
        Arrays.sort(people);
        while(front<=end){
            if(people[front]+people[end]<=limit){
                answer++;
                front++;
                end--;
            }
            else{
                answer++;
                end--;
            }
        }

        return answer;
    }
}

File: programmers/algorithm/level2/Hong0329/구명보트.cpp
//
// Created by hong jun chung on 6/20/24.
//
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;

    sort(people.begin(),people.end(),greater<int>());

    int start=0,end=people.size()-1;

    while(start<end){
        if(people[start]+people[end]<=limit)
        {
            end--;
        }

        answer++;
        start++;
    }
    if(start==end)
        answer++;
    return answer;
}